Swift code guidelines.md

```diff
@@ -0,0 +1,155 @@
+### Swift code guidelines
+
+- [1. Code formatting](#code-formatting)
+- [2. Naming](#naming)
+- [3. Coding Style](#coding-style)
+- [4. Don'ts](#don'ts)
+
+#### 1. Code Formatting
+
+1.1 Use 4 spaces to indent lines.
+1.2. Trim trailing whitespaces (`Xcode` -> `Preferences` -> `Text Editing` -> `Automatically trim trailing whitespace` + `Including whitespace-only lines`).
+1.3. No space before colon and one space after it. Exceptions are the ternary operator `? :`, empty dictionary `[:]` and `#selector` syntax `addTarget(_:action:)`.
+
+**Preferred:**
+```swift
+let fooDictionary: [String: Any]
+let testValue: Double
+func handle(response: Response)
+```
+1.4. No semicolons at the end of line.
+1.5. Use Xcode indentation (`Command-A`, `Command-I`).
+
+#### 2. Naming
+
+2.1. Use `CamelCase` for types and protocols; `lowerCamelCase` for everything else (function, method, property, constant, variable, argument names, enum case).
+2.2. Start factory methods with word `make`.
+2.3. Choose clarity over brevity. 
+2.4. Name booleans like `isAuthenticated` instead of `authenticated`. 
+2.5. Name function parameters. 
```

Comment:
I'd say we should allow for exceptions where appropriate, e.g. where the first argument is plainly related to the function in some way, and where a named argument would make readability suffer.

Reply:
@mikenachbaur-okta Updated.